src/utils/time.ts

```diff
@@ -70,3 +71,41 @@ function quickFormat(timestamp: number, i18next: i18nType, formatRule: string) {
 function locale(i18next: i18nType) {
   return locales[i18next?.language]?.dateFns ?? locales['en-US']?.dateFns
 }
+
+export function formattedAge(interval: Duration) {
+  const years = interval.years ?? 0
+  const months = interval.months ?? 0
+  const days = interval.days ?? 0
+
+  if (years === 0 && months === 0) {
+    if (days === 0) {
+      return i18n.t('time.lessThanADay')
+    } else if (days === 1) {
+      return i18n.t('time.oneDay')
+    } else {
+      return i18n.t('time.days', { days })
+    }
+  } else if (years === 0) {
+    if (months === 1) {
+      return i18n.t('time.oneMonth')
+    } else {
+      return i18n.t('time.months', { months })
+    }
+  } else if (years === 1) {
+    if (months === 0) {
+      return i18n.t('time.oneYear')
+    } else if (months === 1) {
+      return i18n.t('time.oneYearAndOneMonth')
+    } else {
+      return i18n.t('time.oneYearAndMonths', { months })
+    }
+  } else {
+    if (months === 0) {
+      return i18n.t('time.years', { years })
+    } else if (months === 1) {
+      return i18n.t('time.yearsAndOneMonth', { years })
+    } else {
+      return i18n.t('time.yearsAndMonths', { years, months })
+    }
+  }
+}
```

Comment:
This is a pretty meh way of doing this formatting, but it is the only way I could get years and months and localized and days vs day to display correctly. I am open to other options, but also this meets all the design & product requirements 🤷

Reply:
have you considered using https://www.i18next.com/translation-function/context? maybe this can simplify things

Reply:
Thanks for the tip! cleaned this up a bit in 70300b8220c8b3ffb8002f14ab9d1eaa3ba920a2 using context with i18next.